services/logs2notifications/internal/handler/useraction_events.go

```diff
@@ -0,0 +1,288 @@
+}
+
+// There is also a third payload struct called "linkedResource" which we could use, but it seems redundant
+
+type handleUserActionPayload struct {
+	Meta struct {
+		Payload struct {
+			User                   handleUserActionUser    `json:"user"`
+			UserActionEmailDetails *UseractionEmailDetails `json:"userActionEmailDetails,omitempty"`
+		} `json:"payload"`
+	} `json:"meta"`
+}
+
+// These are the basic details that should be piped to the email template
+type UseractionEmailDetails struct {
+	Name             string `json:"name,omitempty"`
+	Email            string `json:"email,omitempty"`
+	Role             string `json:"role,omitempty"`
+	OrganizationName string `json:"organizationName,omitempty"`
+	Keyname          string `json:"keyname,omitempty"`   // for ssh keys
+	Groupname        string `json:"groupname,omitempty"` // for groups
+}
+
+func (h *Messaging) handleUserActionToEmail(notification *Notification, rawPayload []byte) error {
+	var payload handleUserActionPayload
+	if err := json.Unmarshal(rawPayload, &payload); err != nil {
+		return err
+	}
+	// let's check if we need to send an email
+	if payload.Meta.Payload.UserActionEmailDetails == nil {
+		if h.EnableDebug {
+			log.Printf("Skipping email for '%v' event - payload: %v\n", notification.Meta.Event, string(rawPayload))
+		}
+		return nil
+	} else {
+		if h.EnableDebug {
+			log.Printf("Sending email for '%v' event, user:%s\n", notification.Meta.Event, payload.Meta.Payload.UserActionEmailDetails.Email)
+		}
+	}
+
+	useractionEmailDetails := *payload.Meta.Payload.UserActionEmailDetails
+
+	switch notification.Meta.Event {
+	case "api:addUserToOrganization", "api:addAdminToOrganization":
+		return h.addAdminToOrganization(useractionEmailDetails)
+	case "api:removeUserFromOrganization", "api:removeAdminFromOrganization":
+		return h.removeAdminFromOrganization(useractionEmailDetails)
+	case "api:addPlatformRoleToUser":
+		return h.addPlatformRoleToUser(useractionEmailDetails)
+	case "api:removePlatformRoleFromUser":
+		return h.removePlatformRoleFromUser(useractionEmailDetails)
+	case "api:deleteSshKey", "api:deleteSshKeyById":
+		return h.addEditRemoveSshKeyEmailMessage(useractionEmailDetails, notification.Meta.Event, "deleted")
+	case "api:updateSshKey":
+		return h.addEditRemoveSshKeyEmailMessage(useractionEmailDetails, notification.Meta.Event, "updated")
+	case "api:addSshKey":
+		return h.addEditRemoveSshKeyEmailMessage(useractionEmailDetails, notification.Meta.Event, "added")
+	case "api:removeUserFromGroup", "api:addUserToGroup":
+		return h.groupAddRemoveEmailMessage(useractionEmailDetails, notification.Meta.Event)
+	}
+	return errors.New(fmt.Sprintf("Unable to match incoming notification: %v", notification))
+}
+
+func (h *Messaging) groupAddRemoveEmailMessage(valuesStruct UseractionEmailDetails, event string) error {
+
+	content := `
+<p>Hello,</p>
+
+<p>
+  You (<strong>{{.Name}}</strong>) have been {{.Action}} the group "{{.Groupname}}" in your Lagoon account.
```

Comment:
Styling to match others
```suggestion
  You (<strong>{{.Name}}</strong>) have been {{.Action}} the group <strong>{{.Groupname}}</strong>.
```

Reply:
I wonder if its also possible to include which role the user was granted here when the action is adding